tests/robot-tests/scripts/run_tests_pipeline.py

```diff
@@ -0,0 +1,56 @@
+import argparse
+import subprocess
+
+# NOTE(mark): The slack webhook url, and admin and analyst passwords to access to Admin app are
+# stored in the CI pipeline as secret variables, which means they cannot be accessed as normal
+# environment variables, and instead must be passed as an argument to this script.
+
+
+def run_tests_pipeline():
+    assert args.admin_password, "Provide an admin password with an '--admin-pass PASS' argument"
+    assert args.analyst_password, "Provide an analyst password with an '--analyst-pass PASS' argument"
+    assert args.slack_webhook_url, "Please provide slack webhook URL"
+    assert args.env, "Provide an environment with an '--env ENV' argument"
+    assert args.file, "Provide a file/dir to run with an '--file FILE/DIR' argument"
+
+    valid_environments = ['dev', 'test', 'pre-prod', 'prod', 'ci']
+
+    if args.env not in valid_environments:
+        raise Exception(f"Invalid environment provided: {args.env}. Valid environments: {valid_environments}")
+
+    print('Installing dependencies')
+    subprocess.check_call(['google-chrome-stable', '--version'])
+    subprocess.check_call(["python -m", "pip install --upgrade pip"])
+    subprocess.check_call(["pip install", "pipenv"])
+    subprocess.check_call(["pipenv", "install"])
+
+    subprocess.run(
+        f'pipenv run python run_tests.py --admin-pass {args.admin_password} --analyst-pass {args.analyst_password} --slack-webhook-url {args.slack_webhook_url} --env {args.env} --file {args.file} --ci --processes 4 --enable-slack',
+        shell=True)
```

Comment:
At first I wondered whether we could do this in python without resorting to subprocess calls, but having thought about it I think this is fine. I think if we're pragmatic, trying to do this without subprocess calls is more work and/or requires adding several tasks to DevOps. This way is quicker and simpler.

You could possibly argue that we should have just updated the bash script, but I personally think python is less bug prone than bash, so I think this is ok.

That said, others might disagree, so I'll hold back from approving until they've given an opinion.